modules/ldap_groups/app/services/ldap_groups/synchronize_groups_service.rb

```diff
@@ -20,22 +20,25 @@ def call
 
     def synchronize!
       ldap_con = ldap.instance_eval { initialize_ldap_con(account, account_password) }
-      count = 0
 
-      ::LdapGroups::Membership.transaction do
-        @synced_groups.find_each do |sync_group|
-          user_data = get_members(ldap_con, sync_group)
-
-          # Create users that are not existing
-          users = map_to_users(sync_group, user_data)
-
-          update_memberships!(sync_group, users)
-
-          count += users.count
+      @synced_groups.find_each do |sync_group|
+        OpenProject::Mutex.with_advisory_lock_transaction(sync_group) do
+          sleep 10
```

Comment:
Why that `sleep` statement? Is it a leftover from testing the behaviour?

Reply:
Oops, yes :man_facepalming: